net-tools/netwatch/src/netmon.rs

```diff
@@ -30,27 +29,27 @@ use self::actor::{Actor, ActorMessage};
 #[derive(Debug)]
 pub struct Monitor {
     /// Task handle for the monitor task.
-    handle: JoinHandle<()>,
+    #[allow(dead_code)]
+    handle: AbortOnDropHandle<()>,
     actor_tx: mpsc::Sender<ActorMessage>,
 }
 
-impl Drop for Monitor {
-    fn drop(&mut self) {
-        self.handle.abort();
-    }
-}
-
 impl Monitor {
     /// Create a new monitor.
     pub async fn new() -> Result<Self> {
         let actor = Actor::new().await?;
         let actor_tx = actor.subscribe();
 
         let handle = tokio::task::spawn(async move {
-            actor.run().await;
+            if let Err(err) = actor.run().await {
+                warn!("netmon actor died: {:?}", err);
+            }
```

Comment:
This is essentially a supervisor.  The problem with this version is that you get a single WARN in the logfile and things just carry on.  I wonder if maybe it might as well restart the actor - maybe with a small delay.  That is still not proper supervision - if it keeps dying we should bring down the entire Endpoint - but it's a start and we'll flood the logs with WARN messages which could be a bit more obvious.

Reply:
generally yes, but currently there is no error case where this happens (and it is not super easy to restart the actor currently)

Reply:
removed